### PR TITLE
Closes #9150: Do not prompt users for invalid add-on permissions.

### PR DESCRIPTION
**Pull request by Amejia481**
---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/feature/addons/src/main/java/mozilla/components/feature/addons/update/AddonUpdater.kt [hunk 1]
```diff
@@ -186,8 +186,9 @@ class DefaultAddonUpdater(
     ) {
         logger.info("onUpdatePermissionRequest $current")
 
+        val shouldGrantWithoutPrompt = Addon.localizePermissions(newPermissions).isEmpty()
         val wasPreviouslyAllowed =
```
Comment (csadilek) [inline components/feature/addons/src/main/java/mozilla/components/feature/addons/update/AddonUpdater.kt]: nit: would rename this val now. It's no longer just "was previously allowed", maybe "shouldShowNotification" or "shouldShowPrompt"?

File: components/feature/addons/src/test/java/mozilla/components/feature/addons/update/DefaultAddonUpdaterTest.kt [hunk 2]
```diff
@@ -141,6 +141,57 @@ class DefaultAddonUpdaterTest {
         updater.updateStatusStorage.clear(context)
     }
 
+    @Test
+    fun `onUpdatePermissionRequest - should not show a notification for not valid permissions`() {
```
Comment (csadilek) [inline components/feature/addons/src/test/java/mozilla/components/feature/addons/update/DefaultAddonUpdaterTest.kt]: nit: `should not show a notification for not unknown permissions`, or invalid permissions.

**Comment by caugner** [comment]
What issue `#9150` is this PR referring to?

**Comment by Amejia481** [comment]
Sorry https://github.com/mozilla-mobile/fenix/issues/9150

**Pull request merged**